src/Hook.php

```diff
@@ -51,6 +51,13 @@ public static function getSubscribedEvents() : array
     public static function runTypeChecks(Event $composerEvent) : void
     {
         $io       = $composerEvent->getIO();
+
+        if (! file_exists(__DIR__)) {
+            $io->write('<info>' . self::THIS_PACKAGE_NAME . ':</info> Package not found (probably scheduled for removal) - skipping type checks...');
```

Comment:
This seems to be a common pattern for `composer remove`: I would rather report it upstream instead

Reply:
I went to create issue but found this. I guess it's `wont fix` from the upstream.

https://github.com/composer/composer/issues/3729

Reply:
I still think it's an issue: the author self-closed because `composer remove` is indeed a niche feature that is generally avoided by using `composer update` instead, but the bug is there nonetheless.

Reply:
I bring good news https://github.com/composer/composer/issues/8691#issuecomment-598706306

Will be resolved in v2. Therefore, this PR is IMO redundant and can be closed.